cargo/cargo_build_script.bzl

```diff
@@ -114,33 +124,46 @@ def _cargo_build_script_run(ctx, script):
         ),
     ]
 
-def _build_script_impl(ctx):
-    return _cargo_build_script_run(ctx, ctx.executable.script)
-
 _build_script_run = rule(
-    _build_script_impl,
+    doc = (
```

Comment:
I feel it's important to have each rule/macro have it's own documentation. The whole reason I made these changes was because I had to jump around all over the code to figure out what anything was doing. Macros should document what they do as should rules. 

When I wrote 
```
# The source of truth will the `cargo_build_script` macro until stardoc
# implements documentation inheritence. See https://github.com/bazelbuild/stardoc/issues/27
```
I was more trying to inform readers that if the attributes change, then they should look at `cargo_build_script` to ensure the documentation for the wrapper appropriately reflects the changes.

Reply:
"The source of truth..." is perfectly fine with me but should not be a docstring (it's not the documentation of the API, more like a TODO). I don't think we should have the attribute documented in 5 place, this introduce too much toil and leads to wrong documentation at some point there will be a change that update only of those documentation.

Reply:
I agree. I've updated this.

I don't think an attribute should be documented in 5 places either. But I do feel strongly about there being at least some documentation that says:

> Go look at `this_specific_function` for details

This change was removed form this PR though so should continue in the next